Readme.md

```diff
@@ -0,0 +1,171 @@
+
+This repository contains:
+- Reusable workflows: they include the jobs that will be reused by other repositories. These jobs use Github actions to perform the required tasks. All the workflows are available inside a folder named `.github/workflows`. 
+-  Caller templates: templates that references the corresponding reusable workflow. All the templates are available inside a folder named `caller-workflow-templates`. 
+
+The workflows are divided based on the 3 different stages inside the CI/CD pipeline: continuous integration, continuous delivery and continuous deployment. Moreover, each one of these stages also contains workflows for the different distribution/delivery strategies that Graasp repositories use: AWS S3 (or S3 apps) or AWS ECS.
+
+## Features
+### Continuous Integration
+Process within development that continuously builds, tests and merges new code changes. CI allows developers to deploy to the development environment. There, developers can see how new features will work and test improvements without affecting real users. 
+
+This functionality is provided by workflows with the syntax `cintegration-*.yml`. When triggered, three jobs are run: test, build and deploy
+
+### Continuous Delivery
+Automated process that takes the validated code additions from the CI process and relases them to the stage environment. This isolated environment is as similar to the production environment as it can be, and it is where all of the hard core testing will be performed. 
+
+This functionality is provided by workflows with the syntax `cdelivery-*.yml`. When triggered, two jobs are run: build-deploy and test. 
+
+### Continuous Deployment
+When the code is considered to be production-ready, it can be automatically deployed to the production environment.
+
+This functionality is provided by workflows with the syntax `cdeployment-*.yml`. When triggered, one job is run: build-deploy. 
+
+## How to
+### Adding a caller workflow
+Whenever you want to add a specific workflow to your repository, you should copy the appropriate caller template from `caller-workflow-templates` folder in the `.github/workflows` folder of your repository. The template should match the distribution/delivery strategy that your repository uses (AWS S3 or AWS ECS).
+
+Then, the required inputs and secrets should be included/adapted based on the needs of the repository. 
+
+### Triggering a workflow
+Workflow triggers are events that cause a workflow to run. All the Continuous Integration workflows can be manually run from the Actions tab of your repository. 
+
+![Action tab](assets/images/learn-github-actions-repository.png)
+
+Write access to the repository is required to perform these steps.
+
+1. In the left sidebar, click the workflow you want to run.
+
+    ![Action select](assets/images/actions-select-workflow.png)
```

Comment:
Hum I think we can left it as long as it's valid.

Reply:
I refactored and included a deeper explanation on how to trigger the workflows we use in graasp-deploy. I can just leave the link to the Documentation below in case another type of trigger is needed in the future